broker/src/main/java/org/apache/rocketmq/broker/processor/ReplyMessageProcessor.java

```diff
@@ -157,8 +158,10 @@ private PushReplyResult pushReplyMessage(final ChannelHandlerContext ctx,
         final SendMessageRequestHeader requestHeader,
         final Message msg) {
         ReplyMessageRequestHeader replyMessageRequestHeader = new ReplyMessageRequestHeader();
-        replyMessageRequestHeader.setBornHost(ctx.channel().remoteAddress().toString());
-        replyMessageRequestHeader.setStoreHost(this.getStoreHost().toString());
+        InetSocketAddress bornAddress =(InetSocketAddress)(ctx.channel().remoteAddress());
+        replyMessageRequestHeader.setBornHost(bornAddress.getHostName() + ":" + bornAddress.getPort());
```

Comment:
Can we use `getHostString()` here to skip reverse lookup?

Reply:
I'm afraid not. If the service is in the form of IP address + port, the gethoststring method will lose port